src/Twig/Extension/AuthRuntime.php

```diff
@@ -191,6 +191,54 @@ public function getProviders($guid = null)
     }
 
     /**
+     * Return an array of registered OAuth providers for an account with
+     * lst steen and last updated dates
+     *
+     * @param string $guid
+     *
+     * @return array
+     */
+    public function getProvidersDates($guid = null)
+    {
+        $providers = [];
+        if ($guid === null) {
+            $auth = $this->session->getAuthorisation();
+
+            if ($auth === null) {
+              return $providers;
+            }
+            $guid = $auth->getGuid();
+        }
+
+        $providerEntities = $this->records->getProvisionsByGuid($guid);
+        if ($providerEntities === false) {
+            return $providers;
+        }
+
+        /** @var Storage\Entity\Provider $providerEntity */
+        $providers['lastseen'] = null;
+        $providers['lastupdate'] = null;
+        foreach ($providerEntities as $providerEntity) {
+            $provider['name'] = $providerEntity->getProvider();
+            $provider['lastseen'] = $providerEntity->getLastSeen();
+            $provider['lastupdate'] = $providerEntity->getLastUpdate();
+
+            $providers[] = $provider;
+            if($providers['lastseen'] <= $provider['lastseen']) {
+              $providers['lastseen'] = $provider['lastseen'];
+            }
+            if($providers['lastupdate'] <= $provider['lastupdate']) {
+              $providers['lastupdate'] = $provider['lastupdate'];
+            }
+        }
+        if($providers['lastseen'] <= $provider['lastupdate']) {
+          $providers['lastseen'] = $provider['lastupdate'];
```

Comment:
lastseen and lastupdate  are different things though, right? lastseen is when the person last accessed the site, lastupdate is when the account was last updated, for example an account created by an admin but never logged in has never been "seen" but has still been "updated".

This is also inconsistent with how it is handled for other providers on L227-232 where lastseen is not taken as to be the same as lastupdated. Probably better to keep the terms separate.